pkg/controllers/updaterun/controller.go

```diff
@@ -31,9 +32,26 @@ import (
 	"go.goms.io/fleet/pkg/utils"
 	"go.goms.io/fleet/pkg/utils/condition"
 	"go.goms.io/fleet/pkg/utils/controller"
+	"go.goms.io/fleet/pkg/utils/controller/metrics"
 	"go.goms.io/fleet/pkg/utils/informer"
 )
 
+// updateRunMetricsStatus is the status string for update run metrics.
+type updateRunMetricsStatus string
+
+const (
+	// updateRunMetricsStatusInitializing represents the update run is initializing in the metrics.
```

Comment:
are these status in this order?

initializing -> processing -> stuck/completed/failed

Reply:
You are correct.

Reply:
initializing -> failed
initializing -> progressing -> completed/failed
initializing -> progressing -> stuck (per @Arvindthiru's comment below, I think I can delete the progressing metric when stuck is emiited)
initializing -> progressing -> stuck -> progressing -> completed/failed

Reply:
I've updated the comments.